.github/workflows/tests_linux_cpp.yml

```diff
@@ -76,8 +76,9 @@ jobs:
               -DCMAKE_CXX_COMPILER=$(which g++-$GCC_VERSION) \
               -DENABLE_COVERAGE=ON \
               -DLQ_ENABLE_KERNEL_AVX_STREAMING=${{ matrix.enable_kernel_avx_streaming }} \
-              -DLQ_ENABLE_KERNEL_OMP=${{ matrix.enable_kernel_omp }}
-
+              -DLQ_ENABLE_KERNEL_OMP=${{ matrix.enable_kernel_omp }} \
+              -DCATALYST_GIT_TAG="device-interface-rework"
+            # TODO: remove CATALYST_GIT_TAG after release
```

Comment:
Could you please specify which release you are referring to?

Reply:
Also, is this GIT_TAG supposed to be kept up-to-date with the Catalyst main branch?
If not, I'm afraid we'll not stay up-to-date with testing.

Reply:
I don't what your strategy is, @maliasadi ?

Reply:
Our default strategy is to build shared libraries against the "main" branch in Catalyst assuming that the runtime public headers are rarely affected over time. This change will be reverted before merging to master.

Reply:
As discussed offline, let's merge this PR, merge https://github.com/PennyLaneAI/catalyst/pull/1680 and open another PR to remove these temporary git tags.

Just please update the TODO:
```
            # TODO: remove CATALYST_GIT_TAG after merging https://github.com/PennyLaneAI/catalyst/pull/1680
```

Reply:
https://github.com/PennyLaneAI/pennylane-lightning/pull/1143/commits/fffb30fe9ddffa8b6e97eada2eea64ca0cf21aaf